### ISSUE TITLE
Sphereface2 configuration

### ISSUE THREAD

**Issue by GoldMan6** [IDRnD/redimnet#23]
Hi, Great work and thanks for sharing !
i tried to train in redimnetB2 in wespeaker  platform with sphereface2 projection and 15ms hop in features after your recommendations but i get worse results than by training with arc_margin. any idea what might be the problem ?
thanks

**Comment by didi1233** [comment]
> Hi, Great work and thanks for sharing ! i tried to train in redimnetB2 in wespeaker platform with sphereface2 projection and 15ms hop in features after your recommendations but i get worse results than by training with arc_margin. any idea what might be the problem ? thanks

same question

**Comment by vanIvan** [comment]
Hi @GoldMan6 @didi1233 , we know that [issue](https://github.com/IDRnD/redimnet/issues/3), that using wespeaker pipeline as-is it is not possible to achieve same results, but we haven't had an opportunity to dig into this problem. Could you @GoldMan6 please share training config file, you are using

**Comment by GoldMan6** [comment]
Hi @vanIvan, this is the config i used:
num_avg: 10
enable_amp: False # whether enable automatic mixed precision training

seed: 42
num_epochs: 120
save_epoch_interval: 5 # save model every 5 epochs
log_batch_interval: 100 # log every 100 batchs

dataloader_args:
  # batch_size: 256
  batch_size: 200
  num_workers: 16
  pin_memory: false
  prefetch_factor: 4
  drop_last: true

dataset_args:
  # the sample number which will be traversed within one epoch, if the value equals to 0,
  # the utterance number in the dataset will be used as the sample_num_per_epoch.
  sample_num_per_epoch: 0
  shuffle: True
  shuffle_args:
    shuffle_size: 2500
  filter: True
  filter_args:
    min_num_frames: 100
    max_num_frames: 800
  resample_rate: 16000
  speed_perturb: True
  num_frms: 200
  aug_prob: 0.6 # prob to add reverb & noise aug per sample
  fbank_args:
    num_mel_bins: 72
    frame_shift: 15
    frame_length: 25
    dither: 1.0
  spec_aug: False
  spec_aug_args:
    num_t_mask: 1
    num_f_mask: 1
    max_t: 10
    max_f: 8
    prob: 0.6

model: ReDimNetB2
model_init: null
model_args:
  feat_dim: 72
  embed_dim: 192
  pooling_func: "ASTP" # TSTP, ASTP, MQMHASTP
  two_emb_layer: False


projection_args:
  project_type: "sphereface2" # add_margin, arc_margin, sphere, sphereface2, softmax, arc_margin_intertopk_subcenter
  scale: 32.0
  easy_margin: False


margin_scheduler: MarginScheduler
margin_update:
  initial_margin: 0.0
  final_margin: 0.2
  increase_start_epoch: 20
  fix_start_epoch: 40
  update_margin: True
  increase_type: "exp" # exp, linear
  update_margin: true

loss: CrossEntropyLoss
loss_args: {}

optimizer: SGD
optimizer_args:
  momentum: 0.9
  nesterov: True
  weight_decay: 2.0e-05

scheduler: ExponentialDecrease
scheduler_args:
  initial_lr: 0.1
  final_lr: 0.00005
  warm_up_epoch: 6
  warm_from_zero: True

thanks!

**Comment by vanIvan** [comment]
@GoldMan6 thank you, looks correct. Based on this config, everything should be fine, except the fact, that our fork of wespeaker is different in some way from original, in feature extraction, in fact that we have [added](https://github.com/IDRnD/redimnet/issues/3#issuecomment-2322844782) separate argument for head weight decay multiplier. We'll consider sharing our modified wespeaker pipeline in proximate future.

**Comment by GoldMan6** [comment]
Thanks, I will try to  look at the fork, is the difference in the feature extraction also [here](https://github.com/IDRnD/redimnet) ?
Another question, I saw that you've tested the new models you've released on VOICES dataset. can you please share the verification test file you used ? Thanks a lot !
